test/test_hpyerr.py

```diff
@@ -135,6 +135,28 @@ def test_HPyErr_SetObject(self):
             mod.f(ValueError("error message"))
         assert str(err.value) == "error message"
 
+    def test_HPyErr_SetFromErrno(self):
+        import pytest
+        import errno
+        mod = self.make_module("""
+            #include <stdio.h>
+
+            HPyDef_METH(f, "f", f_impl, HPyFunc_O)
+            static HPy f_impl(HPyContext *ctx, HPy self, HPy type)
+            {
+                fseek(stdin, -200, -10);
+                HPyErr_SetFromErrno(ctx, type);
+                return HPy_NULL;
+            }
+            @EXPORT(f)
+            @INIT
+        """)
+        for type in [OSError, IOError]:
+            with pytest.raises(type) as err:
+                mod.f(type)
+
+            assert err.value.errno in [errno.EINVAL, errno.ESPIPE]
```

Comment:
I suppose this is needed because `fseek` could return either error, but I had to read its manpage to understand that. Maybe it's worth a short comment to explain

Reply:
Yes, good point I'll add some explanatory comment. Maybe I'll have to change this altogether, because it does not work on Windows it seems at least from the azure pipelines.